core.ts

```diff
@@ -21,42 +30,119 @@ module.exports = function init(config, callback) {
   _.extend(app.settings, { title: 'BRIK' }, config);
   app.config = config;
 
-  process.title = (config.id || "anonymous") + "-chhaang-" + package.version;
+  process.title =
+    (config.id || 'anonymous') + '-chhaang-' + packageJson.version;
 
   app.log = require('logginator')('chhaang', config.log);
-  require('winston-tagged-http-logger')(server, app.log.createSublogger('http'));
+  require('winston-tagged-http-logger')(
+    server,
+    app.log.createSublogger('http')
+  );
+
+  app.use(cookieParser());
+  app.use(bodyParser.json());
+  app.use(bodyParser.urlencoded({ extended: false }));
 
   // passport & session
   if (config.Feide) {
     var strategy = new SamlStrategy(
       config.Feide.saml || {},
-      function(profile, next) { next(null, profile); });
-    passport.serializeUser(function(user, next) { next(null, user); });
-    passport.deserializeUser(function(user, next) { next(null, user); });
-    passport.use(strategy);
+      function (profile, next) {
+        next(null, profile);
+      }
+    );
+    passport.serializeUser(function (user, next) {
+      next(null, user);
+    });
+    passport.deserializeUser(function (user, next) {
+      next(null, user);
+    });
+    passport.use('saml', strategy);
   }
-  app.use(express.cookieParser());
-  app.use(express.bodyParser());
-  app.use(express.session({ secret: 'sudo apt-get install pants' }));
+  // Open ID
+  else if (config.OpenID) {
+    try {
+      const issuer = await Issuer.discover(config.OpenID.discoveryURL);
+
+      const issuerOptions: ClientMetadata = {
+        client_id: config.OpenID.clientId,
+        client_secret: config.OpenID.clientSecret,
+        redirect_uris: [
+          `${config.OpenID.host}/integration/open-id/login/callback`,
```

Comment:
Actually, I think you could use `config.url` for this. I probably forgot to mention it, and it is incredibly overengineered and undiscoverable anyway, but we have this ["cascading service config"](https://github.com/brikteknologier/cascading-service-config) thing which means the config inherits from a base config, and that base config has a `url` prop which you can use here :D

Reply:
oh right 😻 That's awesome.. I was hoping you had some input here.. So we don't have to configure host for every brikette with open id